core/src/main/java/org/apache/iceberg/io/ResolvingFileIO.java

```diff
@@ -252,4 +252,22 @@ protected void finalize() throws Throwable {
       }
     }
   }
+
+  @Override
+  public Iterable<FileInfo> listPrefix(String prefix) {
+    return withPrefixSupport(io(prefix)).listPrefix(prefix);
+  }
+
+  @Override
+  public void deletePrefix(String prefix) {
+    withPrefixSupport(io(prefix)).deletePrefix(prefix);
+  }
+
+  private SupportsPrefixOperations withPrefixSupport(FileIO io) {
```

Comment:
We don't need this I don't think if we're assuming DelegateFileIO is implemented

Reply:
oops, you're right. This was a leftover